File: semi-labeling/result_viewer/paths.py
from __future__ import annotations

from pathlib import Path


def repo_root() -> Path:
    return Path(__file__).resolve().parents[2]


def lab_root() -> Path:
    return repo_root().parent


def default_source_db() -> Path:
    return lab_root() / "infer_results" / "semi-labeling" / "step2_sematic" / "damage_scan.sqlite3"


def default_feature_db() -> Path:
    return lab_root() / "infer_results" / "semi-labeling" / "step4_feature_grouping" / "feature_groups.sqlite3"


def default_image_root() -> Path:
    return lab_root() / "HinhAnh"

File: app/electron/result_viewer/paths.js
import path from 'node:path';
import { fileURLToPath } from 'node:url';
import fs from 'node:fs';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

export const LABELS = ['crack', 'mold', 'spall'];

export const repoRoot = () => path.resolve(__dirname, '..', '..', '..');

export const labRoot = () => path.dirname(repoRoot());

export const defaultSourceDb = () => path.join(labRoot(), 'infer_results', 'semi-labeling', '2_sematic', 'damage_scan.sqlite3');

export const defaultFeatureDb = () => {
  const sourceDir = path.dirname(defaultSourceDb());
  const giant = path.join(sourceDir, 'step4_feature_grouping_giant_agglo045_full_restart', 'feature_groups.sqlite3');
  if (fs.existsSync(giant)) return giant;
  return path.join(sourceDir, 'step4_feature_grouping', 'feature_groups.sqlite3');
};

export const defaultImageRoot = () => path.join(labRoot(), 'HinhAnh');
